session.go

```diff
@@ -19,31 +19,38 @@ import (
 // This increases the user's time to approve the out-of-band (2nd) factor
 // (e.g. approving a push notification sent to their mobile phone).
 const (
-	MaxLoginRetries         = 7
-	defaultPasswdIterations = 100100
+	MaxLoginRetries = 7
 )
 
-type session struct {
-	// passwdIterations controls how many times password is hashed using PBKDF2
-	// before being sent to LastPass servers.
-	passwdIterations int
-	token            string
-	// user's private key for decrypting sharing keys (encryption keys of shared folders)
-	privateKey *rsa.PrivateKey
+type Session struct {
+	// PasswdIterations controls how many times the user's password
+	// is hashed using PBKDF2 before being sent to LastPass.
+	PasswdIterations int
+
+	// Token is the session token returned by LastPass during
+	// the login process.
+	Token string
+
+	// EncryptionKey is derived by hashing the user's password
+	// using PBKDF2.
+	EncryptionKey []byte
+
+	// OptSharingKey is the user's private key for decrypting sharing
+	// keys. This is used for encryption keys of shared folders.
+	//
+	// This is nil if the user has not generated a sharing key.
+	OptSharingKey *rsa.PrivateKey
```

Comment:
Let's be precise in the naming. It's the private key to decrypt the sharing key. Private key and sharing key are 2 different keys!
I pushed a commit where I rename `OptSharingKey` to `OptPrivateKey`.

Reply:
I guess I don't understand the relationship between those two things very well. My takeaway was that the `PrivateKey` *is* the sharing key or is at least used solely for sharing operations. I guess that could change in the future though?

No worries either way.

Reply:
Every user has (optionally) a private + public key (**asymmetric** encryption). This means user B can encrypt data using the public key of user A, and only user A can decrypt the data using its private key.
Now, instead of encrypting / decrypting all the shared items using asymmetric encryption, user B will encrypt shared folder items with a **symmetric** key which is the sharing key. User B encrypts the sharing key using user A's public key, user A decrypts the sharing key using its private key. User A then decrypts the shared folder items using the sharing key.